### DISCUSSION TITLE
Support device packs for peripheral registers

### DISCUSSION THREAD

**Discussion by chrissnow** [mbed-ce/mbed-os#343] [General]
To be able to debug at low level we need to support SVD files.

the easiest way at the moment is to add to launch.json(which gets overwritten), for example
"svdPath": "Keil::STM32WBxx_DFP@1.4.0",
"deviceName": "STM32WB55RGVx",

We already have deviceName in targets.json (device_name)
svdPath is a bit harder to do nicely since the packs are versioned and it doesn't seem obvious how to get the "latest" pack
Based on this we could add a new field to targets.json, perhaps "cmsis_pack"?

I assume then we need to set some variables in cmake here?
https://github.com/mbed-ce/mbed-os/blob/5354079bfbcd42c22b8437c4f360fbf734ea5d46/tools/cmake/UploadMethodManager.cmake#L56

Then add it launch.json here
https://github.com/mbed-ce/mbed-os/blob/5354079bfbcd42c22b8437c4f360fbf734ea5d46/tools/cmake/mbed_ide_debug_cfg_generator.cmake#L143

Happy to start a PR if this is the right approach.

**Comment by multiplemonomials** [comment]
This would be really awesome, I would love to have the SVD available when debugging.  For the SVD path, you should actually be able to get that information from `targets/cmsis_mcu_descriptions.json` by looking up by the device_name and finding the `from_pack` element.  By all means, make a PR!

**Comment by chrissnow** [comment]
Made a start on this, unfortunately CMAKE's native support for JSON doesn't support JSON5 which means it cant cope with hex numbers in cmsis_mcu_descriptions.json5

```
[cmake] CMake Error at mbed-os/tools/cmake/mbed_ide_debug_cfg_generator.cmake:127 (string):
[cmake]   string sub-command JSON failed parsing json string: * Line 330, Column 26
[cmake] 
[cmake]     Missing ',' or '}' in object declaration
```

**Reply by multiplemonomials** [reply]
ah rip.  Yeah sounds like you need a python script

**Reply by chrissnow** [reply]
Would adding it to tools/python/mbed_tools/build/config.py work?
Could then set the variables in mbed_config.cmake

**Comment by chrissnow** [comment]
Progress so far is here
https://github.com/smartsentry/mbed-os/tree/SVD

if you add the below to your cmakelists it will work
`set(MBED_SVD_NAME "Keil::STM32WBxx_DFP@1.4.0")
`
I'm a bit lost how to get that defined from  cmsis_mcu_descriptions.json5 though...

**Comment by chrissnow** [comment]
@multiplemonomials any thoughts on parsing the JSON5 and setting MBED_SVD_NAME

**Reply by multiplemonomials** [reply]
Here's an example from my memory bank code of interacting with that file
